src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -251,8 +134,8 @@ private static string TryExpandShortFileName(ref StringBuffer outputBuffer, stri
             //
             // Note that we will never get \??\ here as GetFullPathName() does not recognize \??\ and will return it as C:\??\ (or whatever the current drive is).
 
-            int rootLength = PathInternal.GetRootLength(ref outputBuffer);
-            bool isDevice = PathInternal.IsDevice(ref outputBuffer);
+            int rootLength = PathInternal.GetRootLength(outputBuffer);
+            bool isDevice = PathInternal.IsDevice(outputBuffer);
 
             StringBuffer inputBuffer = new StringBuffer(0);
```

Comment:
BTW does it make sense for ValueStringBuilder to grow to absorb the function of STringBuffer ?

Reply:
I called out that I intend to look into that in the PR comments above. It is definitely worth investigating, I just didn't want to make this change too complicated.